src/elements/content-sidebar/SidebarPanels.js

```diff
@@ -68,123 +67,161 @@ const LoadableVersionsSidebar = SidebarUtils.getAsyncSidebarContent(
     MARK_NAME_JS_LOADING_VERSIONS,
 );
 
-const SidebarPanels = ({
-    activitySidebarProps,
-    currentUser,
-    detailsSidebarProps,
-    file,
-    fileId,
-    getPreview,
-    getViewer,
-    hasActivity,
-    hasDetails,
-    hasMetadata,
-    hasSkills,
-    hasVersions,
-    isOpen,
-    metadataSidebarProps,
-    onVersionChange,
-    onVersionHistoryClick,
-    refreshIdentity,
-    versionsSidebarProps,
-}: Props) =>
-    isOpen && (
-        <Switch>
-            {hasSkills && (
-                <Route
-                    exact
-                    path={`/${SIDEBAR_VIEW_SKILLS}`}
-                    render={() => (
-                        <LoadableSkillsSidebar
-                            key={file.id}
-                            file={file}
-                            getPreview={getPreview}
-                            getViewer={getViewer}
-                            startMarkName={MARK_NAME_JS_LOADING_SKILLS}
+class SidebarPanels extends React.Component<Props, State> {
+    activitySidebarRef: React.Ref<any> = React.createRef();
```

Comment:
these refs are quite repetitive; I'm thinking of adding a constructor so I can iterate over the constants and create the refs dynamically instead

Reply:
I personally prefer it this way. It's explicit and clear what's going on.